google/cloud/bigtable/internal/async_bulk_apply_test.cc

```diff
@@ -605,10 +764,29 @@ TEST_F(AsyncBulkApplyTest, CancelMidStream) {
                              IdempotentMutation("r2"));
   promise<absl::optional<v2::MutateRowsResponse>> p;
 
+#ifdef GOOGLE_CLOUD_CPP_BIGTABLE_WITH_OTEL_METRICS
+  auto mock_metric = std::make_unique<MockMetric>();
+  EXPECT_CALL(*mock_metric, PreCall).Times(1);
+  EXPECT_CALL(*mock_metric, PostCall).Times(1);
+  EXPECT_CALL(*mock_metric, OnDone).Times(1);
+
+  auto fake_metric = std::make_shared<CloningMetric>(std::move(mock_metric));
+  auto clock = std::make_shared<testing_util::FakeSteadyClock>();
+
+  // Normally std::make_shared would be used here, but some weird type deduction
+  // is preventing it.
+  // NOLINTNEXTLINE(modernize-make-shared)
+  auto operation_context = std::shared_ptr<OperationContext>(
+      new OperationContext({}, {}, {fake_metric}, clock));
+#else
+  auto operation_context = std::make_shared<OperationContext>();
+#endif
```

Comment:
Same as above, shall we split these test cases into two (w/ and w/o otel) to keep the testing behavior-driven?